src/nti/fakestatsd/metric.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, absolute_import, division
+__docformat__ = "restructuredtext en"
+
+
+METRIC_GAUGE_KIND = 'g'
+METRIC_COUNTER_KIND = 'c'
+METRIC_SET_KIND = 's'
+METRIC_TIMER_KIND = 'ms'
+
+
+def _parse_sampling_data(data):
+    if not data.startswith('@'):
+        raise ValueError('Expected "@" in sampling data. %s' % data)
+    return float(data[1:])
+
+def _as_metric(metric_data):
+    """
+    Parses a single metric packet in to `Metric`.
+    
+    Metrics take the form of <name>:<value>|<type>(|@<sampling_rate>)
+
+    A ValueError is raised for invalid data
+    """
+
+    sampling = None
+    name = None
+    value = None
+    kind = None
+    parts = metric_data.split('|')
+    if len(parts) < 2 or len(parts) > 3:
+        raise ValueError('Unexpected metric data %s. Wrong number of parts' % metric_data)
+
+    if len(parts) == 3:
+        sampling_data = parts.pop(-1)
+        sampling = _parse_sampling_data(sampling_data)
+
+    kind = parts[1]
+    name, value = parts[0].split(':')
+
+    return Metric(name, value, kind, sampling_rate=sampling)
+
+def _as_metrics(data):
+    """
+    Parses the statsd data packet to a _list_ of metrics.
+
+    See also: https://github.com/etsy/statsd/blob/master/docs/metric_types.md
+    """
+    metrics = []
+
+    # Multi metric packets are seperated by newlines
+    for metric_data in data.split('\n'):
+        metrics.append(_as_metric(metric_data))
+    return metrics
+
+
+class Metric(object):
+    """
+    The representation of a single statsd metric.
+
+    Attributes:
```

Comment:
You can document these using the `.. attribute:: name` directive.
```python
"""
Some class

.. attribute:: name
    The metric name
.. attribute:: value
    A value
"""
```

You can also document them by setting defaults on the class and using the special `#:` comment syntax:
```python
class Foo(object):

    #: The metric name
    name = None
```